app/models/kpi.rb

```diff
@@ -0,0 +1,366 @@
+class Kpi < CouchRest::Model::Base
+  include Record
+
+  INTRODUCTION_AND_ENGAGEMENT_TYPE = {
+    name: 'introduction_and_engagement',
+    calculations: ['rate_of_report', 'elapsed_time_for_incident_and_report', 'average_delay_of_access'],
+    days_ranges: [
+      [0, 3],
+      [4, 5],
+      [6, 14],
+      [15, 30],
+      [31, 92]
+    ]
+  }
+
+  ASSESSMENT_TYPE = {
+    name: 'assessment',
+    calculations: ['completed_asssessment', 'approved_assessment']
+  }	
+    CASE_ACTION_PLANNING_TYPE = {
+    name: 'case_action_planning',
+    calculations: ['needed_safety_plan', 'completed_case_action_plan', 'approved_case_action_plan']
+  }
+
+  IMPLEMENT_THE_CASE_ACTION_PLAN_TYPE = {
+    name: 'implement_the_case_action_plan',
+    calculations: ['services_provided']
+  }
+
+  CASE_FOLLOW_UP_TYPE = {
+    name: 'case_follow_up',
+    calculations: ['follow_up_meetings', 'average_follow_up_meetings']
+  }
+
+  CASE_CLOSURE_TYPE = {
+    name: 'case_closure',
+    calculations: [
+      'elapsed_time_from_opening_to_closure',
+      'elapsed_time_from_opening_to_closure_hight_risk',
+      'reason_for_case_closure',
+      'case_closure_rate'
+    ],
+    days_ranges: [
+      [0, 30],
+      [31, 92],
+      [93, 185],
+    ]
+  }
+
+  CLIENT_SATISFACTION_TYPE = {
+    name: 'client_satisfaction',
+    calculations: ['client_satisfaction']
+  }
+
+  OTHER_TYPE = {
+    name: 'other_kpi',
+    calculations: [
+      'case_load',
+      'supervisor_to_case_worker_ratio',
+      'caseworkers_active_on_gbvims_plus'
+    ],
+  }
+
+  TYPES = [
+    INTRODUCTION_AND_ENGAGEMENT_TYPE,
+    ASSESSMENT_TYPE,
+    CASE_ACTION_PLANNING_TYPE,
+    IMPLEMENT_THE_CASE_ACTION_PLAN_TYPE,
+    CASE_FOLLOW_UP_TYPE,
+    CASE_CLOSURE_TYPE,
+    CLIENT_SATISFACTION_TYPE,
+    OTHER_TYPE,
+  ]
+
+  NEEDED_DATE_RANGE = ['follow_up_meetings', 'completed_asssessment']
+
+  def self.minimum_reportable_fields
```

Comment:
Prefer to have a  "class << self" block instead of having these "def self.<something" method definitions.

```suggestion
  class << self
    def minimum_reportable_fields
       <method lines>
    end
     
    def get_default_range_from(range)
       <method lines>
    end 
     ...  
  end
```